datalad/support/dataset.py

```diff
@@ -0,0 +1,60 @@
+# emacs: -*- mode: python-mode; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""
+This layer makes the difference between an arbitrary annex and a datalad-managed dataset.
+
+"""
+__author__ = 'Benjamin Poldrack'
+
+from os.path import join
+
+from annexrepo import AnnexRepo
+
+class Dataset(AnnexRepo):
+    """Representation of a dataset handled by datalad.
+
+    Implementations of datalad commands are supposed to use this rather than AnnexRepo or GitRepo directly,
+    since any restrictions on annexes required by datalad due to its cross-platform distribution approach are handled
+    within this class. Also an AnnexRepo has no idea of any datalad configuration needs, of course.
+
+    """
+
+    def __init__(self, path, url=None):
+        """Creates a dataset representation from path.
+
+        If `path` is empty, it creates an new repository.
+        If `url` is given, it is expected to point to a git repository to create a clone from.
+
+        Parameters
+        ----------
+
+        path : str
+               path to repository
+
```

Comment:
They do. Well, I'll try to figure out, what you "usually do". :)

2015-02-25 15:04 GMT+01:00 Yaroslav Halchenko <notifications@github.com>:

> In datalad/support/dataset.py
> <https://github.com/datalad/datalad/pull/42#discussion_r25341265>:
>
> > +    within this class. Also an AnnexRepo has no idea of any datalad configuration needs, of course.
> > +
> > +    """
> > +
> > +    def __init__(self, path, url=None):
> > +        """Creates a dataset representation from path.
> > +
> > +        If `path` is empty, it creates an new repository.
> > +        If `url` is given, it is expected to point to a git repository to create a clone from.
> > +
> > +        Parameters
> > +        ----------
> > +
> > +        path : str
> > +               path to repository
> > +
>
> do they demand empty lines around entries? we usually don't do that so
> even if they do -- let's stay consistent and not add empty lines
> before/between entries
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/datalad/datalad/pull/42/files#r25341265>.
>